lib/server/monalisa-synchronization/MonAlisaSynchronizer.js

```diff
@@ -54,31 +54,49 @@ class MonAlisaSynchronizer {
             const dataPasses = (await this._monAlisaInterface.getDataPasses())
                 .filter((monalisaDataPass) => this._doesDataPassNeedUpdate(monalisaDataPass, dataPassesLastRunNumber));
 
-            // eslint-disable-next-line require-jsdoc
-            const transactionablePipeline = async (monalisaDataPass) => {
-                const lhcPeriodDB = await getOrCreateLhcPeriod({ name: extractLhcPeriod(monalisaDataPass.name).name });
-                await DataPassRepository.upsert({ lhcPeriodId: lhcPeriodDB.id, ...monalisaDataPass });
-                const dataPass = await DataPassRepository.findOne({ where: { name: monalisaDataPass.name } });
-                const { runNumbers } = await this._monAlisaInterface.getDataPassDetails(dataPass.description);
-                const associatedRuns = await RunRepository.findAll({
-                    where: {
-                        runNumber: { [Op.in]: runNumbers },
-                        definition: RunDefinition.Physics,
-                    },
-                });
-                if (associatedRuns.length > 0) {
-                    await dataPass.addRuns(associatedRuns, { ignoreDuplicates: true });
+            const errors = {};
+            for (const monalisaDataPass of dataPasses) {
+                try {
+                    await dataSource.transaction(() => this._updateOneDataPass(monalisaDataPass));
+                } catch (error) {
+                    if (!errors[error.message]) {
+                        errors[error.message] = [];
+                    }
+                    errors[error.message].push(monalisaDataPass.name);
                 }
-            };
+            }
+            if (Object.entries(errors).length !== 0) {
+                const stackedError = Object.entries(errors)
+                    .map(([errorMessage, dataPassesList]) => `Error for data passes: ${dataPassesList.join(',')}\n   ${errorMessage}`)
```

Comment:
dataPassesList => dataPassesNames once again, naming is very important

Reply:
removed